light_curves/code_src/hats_functions.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+import pandas as pd
+from astropy.table import Table
+import lsdb
+from dask.distributed import Client
+from data_structures import MultiIndexDFObject
+from upath import UPath
+
+def hats_get_lightcurves(
+    sample_table: Table,
+    object_catalog_path: str,
+    lightcurve_catalog_path: str,
+    object_columns: list[str],
+    light_curve_columns: list[str],
+    filter_id_to_name: dict[int, str],
+    id_col: str,
+    time_col: str,
+    flux_col: str,
+    err_col: str,
+    *,
```

Comment:
```suggestion
    *,
    object_catalog_path: str,
    lightcurve_catalog_path: str,
    object_columns: list[str],
    light_curve_columns: list[str],
    filter_id_to_name: dict[int, str],
    id_col: str,
    time_col: str,
    flux_col: str,
    err_col: str,
```

I would make all of these keyword-only since it would be easy to mix them up.